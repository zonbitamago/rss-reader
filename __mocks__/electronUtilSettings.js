export const electronUtil = {
  closeApp: () => {
    console.log("closeApp");
  },
  minimizeApp: () => {
    console.log("minimizeApp");
  },
  openBrowser: url => {
    console.log("openBrowser");
    console.log(url);
  }
};
