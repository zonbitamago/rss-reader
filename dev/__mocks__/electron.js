module.exports = {
  ipcRenderer: {
    send: jest.fn()
  },
  remote: {
    app: {
      getPath: () => {
        return "";
      }
    }
  }
};
