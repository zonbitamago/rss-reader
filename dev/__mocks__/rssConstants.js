export const RSS1 = `
    <?xml version="1.0" encoding="utf-8" ?>
<rdf:RDF xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns="http://purl.org/rss/1.0/">
<channel rdf:about="http://syncer.jp/about"><title>Syncer</title>
<link>http://syncer.jp/</link>
<description><![CDATA[知識と感動を同期(Sync)するブログ]]></description>
<items>
<rdf:Seq>
<rdf:li resource="http://syncer.jp/how-to-make-feed-by-php"/>
</rdf:Seq>
</items>
</channel>
<item rdf:about="http://syncer.jp/how-to-make-feed-by-php">
<title>PHPでRSS、AtomのFeedを作成する方法</title>
<link>http://syncer.jp/how-to-make-feed-by-php</link>
<dc:date>2014-11-23</dc:date>
<description><![CDATA[PHPを使って、RSS、Atomのフィード・ファイルを作成する方法を解説します。]]></description>
<dc:subject>ここの値を指定</dc:subject>
</item>
<item rdf:about="http://syncer.jp/how-to-make-feed-by-php">
<title>PHPでRSS、AtomのFeedを作成する方法</title>
<link>http://syncer.jp/how-to-make-feed-by-php</link>
<dc:date>2014-11-23</dc:date>
<description><![CDATA[PHPを使って、RSS、Atomのフィード・ファイルを作成する方法を解説します。]]></description>
<dc:subject>ここの値を指定</dc:subject>
</item>
</rdf:RDF>
    `;

export const RSS2 = `
            <?xml version="1.0"?>
<rss version="2.0">
   <channel>
      <title>Liftoff News</title>
      <link>http://liftoff.msfc.nasa.gov/</link>
      <description>Liftoff to Space Exploration.</description>
      <language>en-us</language>
      <pubDate>Tue, 10 Jun 2003 04:00:00 GMT</pubDate>
      <lastBuildDate>Tue, 10 Jun 2003 09:41:01 GMT</lastBuildDate>
      <docs>http://blogs.law.harvard.edu/tech/rss</docs>
      <generator>Weblog Editor 2.0</generator>
      <managingEditor>editor@example.com</managingEditor>
      <webMaster>webmaster@example.com</webMaster>
      <item>
         <title>Star City</title>
         <author>writer@example.com (Writer)</author>
         <link>http://liftoff.msfc.nasa.gov/news/2003/news-starcity.asp</link>
         <description>How do Americans get ready to work with Russians aboard the International Space Station? They take a crash course in culture, language and protocol at Russia's &lt;a href="http://howe.iki.rssi.ru/GCTC/gctc_e.htm"&gt;Star City&lt;/a&gt;.</description>
         <pubDate>Tue, 03 Jun 2003 09:39:21 GMT</pubDate>
         <guid>http://liftoff.msfc.nasa.gov/2003/06/03.html#item573</guid>
      </item>
      <item>
         <description>Sky watchers in Europe, Asia, and parts of Alaska and Canada will experience a &lt;a href="http://science.nasa.gov/headlines/y2003/30may_solareclipse.htm"&gt;partial eclipse of the Sun&lt;/a&gt; on Saturday, May 31st.</description>
         <pubDate>Fri, 30 May 2003 11:06:42 GMT</pubDate>
         <guid>http://liftoff.msfc.nasa.gov/2003/05/30.html#item572</guid>
      </item>
      <item>
         <title>The Engine That Does More</title>
         <link>http://liftoff.msfc.nasa.gov/news/2003/news-VASIMR.asp</link>
         <description>Before man travels to Mars, NASA hopes to design new engines that will let us fly through the Solar System more quickly.  The proposed VASIMR engine would do that.</description>
         <pubDate>Tue, 27 May 2003 08:37:32 GMT</pubDate>
         <guid>http://liftoff.msfc.nasa.gov/2003/05/27.html#item571</guid>
      </item>
      <item>
         <title>Astronauts' Dirty Laundry</title>
         <link>http://liftoff.msfc.nasa.gov/news/2003/news-laundry.asp</link>
         <description>Compared to earlier spacecraft, the International Space Station has many luxuries, but laundry facilities are not one of them.  Instead, astronauts have other options.</description>
         <pubDate>Tue, 20 May 2003 08:56:02 GMT</pubDate>
         <guid>http://liftoff.msfc.nasa.gov/2003/05/20.html#item570</guid>
      </item>
   </channel>
</rss>
`;

export const ATOM = `
<?xml version="1.0" encoding="utf-8"?>
<feed xmlns="http://www.w3.org/2005/Atom">
        <title>Example Feed</title>
        <subtitle>Insert witty or insightful remark here</subtitle>
        <link href="http://example.org/"/>
        <updated>2003-12-13T18:30:02Z</updated>
        <author>
                <name>John Doe</name>
                <email>johndoe@example.com</email>
        </author>
        <id>urn:uuid:60a76c80-d399-11d9-b93C-0003939e0af6</id>
        <entry>
                <title>Atom-Powered Robots Run Amok</title>
                <link href="http://example.org/2003/12/13/atom03"/>
                <id>urn:uuid:1225c695-cfb8-4ebb-aaaa-80da344efa6a</id>
                <updated>2003-12-13T18:30:02Z</updated>
                <summary>Some text.</summary>
        </entry>
        <entry>
                <title>Atom-Powered Robots Run Amok</title>
                <link href="http://example.org/2003/12/13/atom03"/>
                <id>urn:uuid:1225c695-cfb8-4ebb-aaaa-80da344efa6a</id>
                <updated>2003-12-13T18:30:02Z</updated>
                <summary>Some text.</summary>
        </entry>
</feed>
`;

export const RSS_FEED = [
  {
    author: "Writer",
    categories: [],
    comments: null,
    date: "2003-06-03T09:39:21.000Z",
    description:
      'How do Americans get ready to work with Russians aboard the International Space Station? They take a crash course in culture, language and protocol at Russia\'s <a href=\\"http://howe.iki.rssi.ru/GCTC/gctc_e.htm\\">Star City</a>.',
    enclosures: [],
    guid: "http://liftoff.msfc.nasa.gov/2003/06/03.html#item573",
    image: {},
    link: "http://liftoff.msfc.nasa.gov/news/2003/news-starcity.asp",
    meta: {
      "#ns": [],
      "#type": "rss",
      "#version": "2.0",
      "#xml": {
        version: "1.0"
      },
      "@": [],
      author: "editor@example.com",
      categories: [],
      cloud: {},
      copyright: null,
      date: "2003-06-10T09:41:01.000Z",
      description: "Liftoff to Space Exploration.",
      favicon: null,
      generator: "Weblog Editor 2.0",
      image: {},
      language: "en-us",
      link: "http://liftoff.msfc.nasa.gov/",
      pubDate: "2003-06-10T04:00:00.000Z",
      pubdate: "2003-06-10T04:00:00.000Z",
      "rss:@": {},
      "rss:description": {
        "#": "Liftoff to Space Exploration.",
        "@": {}
      },
      "rss:docs": {
        "#": "http://blogs.law.harvard.edu/tech/rss",
        "@": {}
      },
      "rss:generator": {
        "#": "Weblog Editor 2.0",
        "@": {}
      },
      "rss:language": {
        "#": "en-us",
        "@": {}
      },
      "rss:lastbuilddate": {
        "#": "Tue, 10 Jun 2003 09:41:01 GMT",
        "@": {}
      },
      "rss:link": {
        "#": "http://liftoff.msfc.nasa.gov/",
        "@": {}
      },
      "rss:managingeditor": {
        "#": "editor@example.com",
        "@": {},
        email: "editor@example.com",
        name: ""
      },
      "rss:pubdate": {
        "#": "Tue, 10 Jun 2003 04:00:00 GMT",
        "@": {}
      },
      "rss:title": {
        "#": "Liftoff News",
        "@": {}
      },
      "rss:webmaster": {
        "#": "webmaster@example.com",
        "@": {},
        email: "webmaster@example.com",
        name: ""
      },
      title: "Liftoff News",
      xmlUrl: null,
      xmlurl: null
    },
    origlink: null,
    permalink: "http://liftoff.msfc.nasa.gov/2003/06/03.html#item573",
    pubDate: "2003-06-03T09:39:21.000Z",
    pubdate: "2003-06-03T09:39:21.000Z",
    "rss:@": {},
    "rss:author": {
      "#": "writer@example.com (Writer)",
      "@": {},
      email: "writer@example.com",
      name: "Writer"
    },
    "rss:description": {
      "#":
        'How do Americans get ready to work with Russians aboard the International Space Station? They take a crash course in culture, language and protocol at Russia\'s <a href=\\"http://howe.iki.rssi.ru/GCTC/gctc_e.htm\\">Star City</a>.',
      "@": {}
    },
    "rss:guid": {
      "#": "http://liftoff.msfc.nasa.gov/2003/06/03.html#item573",
      "@": {}
    },
    "rss:link": {
      "#": "http://liftoff.msfc.nasa.gov/news/2003/news-starcity.asp",
      "@": {}
    },
    "rss:pubdate": {
      "#": "Tue, 03 Jun 2003 09:39:21 GMT",
      "@": {}
    },
    "rss:title": {
      "#": "Star City",
      "@": {}
    },
    source: {},
    summary:
      'How do Americans get ready to work with Russians aboard the International Space Station? They take a crash course in culture, language and protocol at Russia\'s <a href=\\"http://howe.iki.rssi.ru/GCTC/gctc_e.htm\\">Star City</a>.',
    title: "Star City"
  },
  {
    author: "editor@example.com",
    categories: [],
    comments: null,
    date: "2003-05-31T11:06:42.000Z",
    description:
      'Sky watchers in Europe, Asia, and parts of Alaska and Canada will experience a <a href=\\"http://science.nasa.gov/headlines/y2003/30may_solareclipse.htm\\">partial eclipse of the Sun</a> on Saturday, May 31st.',
    enclosures: [],
    guid: "http://liftoff.msfc.nasa.gov/2003/05/30.html#item572",
    image: {},
    link: "http://liftoff.msfc.nasa.gov/2003/05/30.html#item572",
    meta: {
      "#ns": [],
      "#type": "rss",
      "#version": "2.0",
      "#xml": {
        version: "1.0"
      },
      "@": [],
      author: "editor@example.com",
      categories: [],
      cloud: {},
      copyright: null,
      date: "2003-06-10T09:41:01.000Z",
      description: "Liftoff to Space Exploration.",
      favicon: null,
      generator: "Weblog Editor 2.0",
      image: {},
      language: "en-us",
      link: "http://liftoff.msfc.nasa.gov/",
      pubDate: "2003-06-10T04:00:00.000Z",
      pubdate: "2003-06-10T04:00:00.000Z",
      "rss:@": {},
      "rss:description": {
        "#": "Liftoff to Space Exploration.",
        "@": {}
      },
      "rss:docs": {
        "#": "http://blogs.law.harvard.edu/tech/rss",
        "@": {}
      },
      "rss:generator": {
        "#": "Weblog Editor 2.0",
        "@": {}
      },
      "rss:language": {
        "#": "en-us",
        "@": {}
      },
      "rss:lastbuilddate": {
        "#": "Tue, 10 Jun 2003 09:41:01 GMT",
        "@": {}
      },
      "rss:link": {
        "#": "http://liftoff.msfc.nasa.gov/",
        "@": {}
      },
      "rss:managingeditor": {
        "#": "editor@example.com",
        "@": {},
        email: "editor@example.com",
        name: ""
      },
      "rss:pubdate": {
        "#": "Tue, 10 Jun 2003 04:00:00 GMT",
        "@": {}
      },
      "rss:title": {
        "#": "Liftoff News",
        "@": {}
      },
      "rss:webmaster": {
        "#": "webmaster@example.com",
        "@": {},
        email: "webmaster@example.com",
        name: ""
      },
      title: "Liftoff News",
      xmlUrl: null,
      xmlurl: null
    },
    origlink: null,
    permalink: "http://liftoff.msfc.nasa.gov/2003/05/30.html#item572",
    pubDate: "2003-05-31T11:06:42.000Z",
    pubdate: "2003-05-31T11:06:42.000Z",
    "rss:@": {},
    "rss:description": {
      "#":
        'Sky watchers in Europe, Asia, and parts of Alaska and Canada will experience a <a href=\\"http://science.nasa.gov/headlines/y2003/30may_solareclipse.htm\\">partial eclipse of the Sun</a> on Saturday, May 31st.',
      "@": {}
    },
    "rss:guid": {
      "#": "http://liftoff.msfc.nasa.gov/2003/05/30.html#item572",
      "@": {}
    },
    "rss:pubdate": {
      "#": "Fri, 30 May 2003 11:06:42 GMT",
      "@": {}
    },
    source: {},
    summary:
      'Sky watchers in Europe, Asia, and parts of Alaska and Canada will experience a <a href=\\"http://science.nasa.gov/headlines/y2003/30may_solareclipse.htm\\">partial eclipse of the Sun</a> on Saturday, May 31st.',
    title: null
  },
  {
    author: "editor@example.com",
    categories: [],
    comments: null,
    date: "2003-05-27T08:37:32.000Z",
    description:
      "Before man travels to Mars, NASA hopes to design new engines that will let us fly through the Solar System more quickly.  The proposed VASIMR engine would do that.",
    enclosures: [],
    guid: "http://liftoff.msfc.nasa.gov/2003/05/27.html#item571",
    image: {},
    link: "http://liftoff.msfc.nasa.gov/news/2003/news-VASIMR.asp",
    meta: {
      "#ns": [],
      "#type": "rss",
      "#version": "2.0",
      "#xml": {
        version: "1.0"
      },
      "@": [],
      author: "editor@example.com",
      categories: [],
      cloud: {},
      copyright: null,
      date: "2003-06-10T09:41:01.000Z",
      description: "Liftoff to Space Exploration.",
      favicon: null,
      generator: "Weblog Editor 2.0",
      image: {},
      language: "en-us",
      link: "http://liftoff.msfc.nasa.gov/",
      pubDate: "2003-06-10T04:00:00.000Z",
      pubdate: "2003-06-10T04:00:00.000Z",
      "rss:@": {},
      "rss:description": {
        "#": "Liftoff to Space Exploration.",
        "@": {}
      },
      "rss:docs": {
        "#": "http://blogs.law.harvard.edu/tech/rss",
        "@": {}
      },
      "rss:generator": {
        "#": "Weblog Editor 2.0",
        "@": {}
      },
      "rss:language": {
        "#": "en-us",
        "@": {}
      },
      "rss:lastbuilddate": {
        "#": "Tue, 10 Jun 2003 09:41:01 GMT",
        "@": {}
      },
      "rss:link": {
        "#": "http://liftoff.msfc.nasa.gov/",
        "@": {}
      },
      "rss:managingeditor": {
        "#": "editor@example.com",
        "@": {},
        email: "editor@example.com",
        name: ""
      },
      "rss:pubdate": {
        "#": "Tue, 10 Jun 2003 04:00:00 GMT",
        "@": {}
      },
      "rss:title": {
        "#": "Liftoff News",
        "@": {}
      },
      "rss:webmaster": {
        "#": "webmaster@example.com",
        "@": {},
        email: "webmaster@example.com",
        name: ""
      },
      title: "Liftoff News",
      xmlUrl: null,
      xmlurl: null
    },
    origlink: null,
    permalink: "http://liftoff.msfc.nasa.gov/2003/05/27.html#item571",
    pubDate: "2003-05-27T08:37:32.000Z",
    pubdate: "2003-05-27T08:37:32.000Z",
    "rss:@": {},
    "rss:description": {
      "#":
        "Before man travels to Mars, NASA hopes to design new engines that will let us fly through the Solar System more quickly.  The proposed VASIMR engine would do that.",
      "@": {}
    },
    "rss:guid": {
      "#": "http://liftoff.msfc.nasa.gov/2003/05/27.html#item571",
      "@": {}
    },
    "rss:link": {
      "#": "http://liftoff.msfc.nasa.gov/news/2003/news-VASIMR.asp",
      "@": {}
    },
    "rss:pubdate": {
      "#": "Tue, 27 May 2003 08:37:32 GMT",
      "@": {}
    },
    "rss:title": {
      "#": "The Engine That Does More",
      "@": {}
    },
    source: {},
    summary:
      "Before man travels to Mars, NASA hopes to design new engines that will let us fly through the Solar System more quickly.  The proposed VASIMR engine would do that.",
    title: "The Engine That Does More"
  },
  {
    author: "editor@example.com",
    categories: [],
    comments: null,
    date: "2003-05-20T08:56:02.000Z",
    description:
      "Compared to earlier spacecraft, the International Space Station has many luxuries, but laundry facilities are not one of them.  Instead, astronauts have other options.",
    enclosures: [],
    guid: "http://liftoff.msfc.nasa.gov/2003/05/20.html#item570",
    image: {},
    link: "http://liftoff.msfc.nasa.gov/news/2003/news-laundry.asp",
    meta: {
      "#ns": [],
      "#type": "rss",
      "#version": "2.0",
      "#xml": {
        version: "1.0"
      },
      "@": [],
      author: "editor@example.com",
      categories: [],
      cloud: {},
      copyright: null,
      date: "2003-06-10T09:41:01.000Z",
      description: "Liftoff to Space Exploration.",
      favicon: null,
      generator: "Weblog Editor 2.0",
      image: {},
      language: "en-us",
      link: "http://liftoff.msfc.nasa.gov/",
      pubDate: "2003-06-10T04:00:00.000Z",
      pubdate: "2003-06-10T04:00:00.000Z",
      "rss:@": {},
      "rss:description": {
        "#": "Liftoff to Space Exploration.",
        "@": {}
      },
      "rss:docs": {
        "#": "http://blogs.law.harvard.edu/tech/rss",
        "@": {}
      },
      "rss:generator": {
        "#": "Weblog Editor 2.0",
        "@": {}
      },
      "rss:language": {
        "#": "en-us",
        "@": {}
      },
      "rss:lastbuilddate": {
        "#": "Tue, 10 Jun 2003 09:41:01 GMT",
        "@": {}
      },
      "rss:link": {
        "#": "http://liftoff.msfc.nasa.gov/",
        "@": {}
      },
      "rss:managingeditor": {
        "#": "editor@example.com",
        "@": {},
        email: "editor@example.com",
        name: ""
      },
      "rss:pubdate": {
        "#": "Tue, 10 Jun 2003 04:00:00 GMT",
        "@": {}
      },
      "rss:title": {
        "#": "Liftoff News",
        "@": {}
      },
      "rss:webmaster": {
        "#": "webmaster@example.com",
        "@": {},
        email: "webmaster@example.com",
        name: ""
      },
      title: "Liftoff News",
      xmlUrl: null,
      xmlurl: null
    },
    origlink: null,
    permalink: "http://liftoff.msfc.nasa.gov/2003/05/20.html#item570",
    pubDate: "2003-05-20T08:56:02.000Z",
    pubdate: "2003-05-20T08:56:02.000Z",
    "rss:@": {},
    "rss:description": {
      "#":
        "Compared to earlier spacecraft, the International Space Station has many luxuries, but laundry facilities are not one of them.  Instead, astronauts have other options.",
      "@": {}
    },
    "rss:guid": {
      "#": "http://liftoff.msfc.nasa.gov/2003/05/20.html#item570",
      "@": {}
    },
    "rss:link": {
      "#": "http://liftoff.msfc.nasa.gov/news/2003/news-laundry.asp",
      "@": {}
    },
    "rss:pubdate": {
      "#": "Tue, 20 May 2003 08:56:02 GMT",
      "@": {}
    },
    "rss:title": {
      "#": "Astronauts' Dirty Laundry",
      "@": {}
    },
    source: {},
    summary:
      "Compared to earlier spacecraft, the International Space Station has many luxuries, but laundry facilities are not one of them.  Instead, astronauts have other options.",
    title: "Astronauts' Dirty Laundry"
  }
];
