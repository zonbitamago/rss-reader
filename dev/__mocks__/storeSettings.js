export let store = {
  ItemStore: { items: [], add: () => console.log("add") },
  SettingStore: {
    updateDuration: 5,
    getSettings: () => console.log("getSettings"),
    setSetting: () => console.log("setSetting")
  },
  RssListStore: {
    rssList: [
      { name: "google", url: "https://www.google.co.jp/" },
      { name: "yahoo", url: "https://www.yahoo.co.jp/" }
    ],
    getRssList: () => console.log("getRssList"),
    setRssList: () => console.log("setRssList"),
    deleteRssList: () => console.log("deleteRssList")
  }
};
