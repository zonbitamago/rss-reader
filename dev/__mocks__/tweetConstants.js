export const TWEET_LIST = {
  created_at: "Sat Jun 23 05: 24: 50 +0000 2018",
  id: 1010393219274698800,
  id_str: "1010393219274698752",
  text: "RT @eri_sai: 息子よ...問題を読みなさい https: //t.co/qedW1jUMRK",
  truncated: false,
  entities: {
    hashtags: [],
    symbols: [],
    user_mentions: [[Object]],
    urls: [],
    media: [[Object]]
  },
  extended_entities: {
    media: [[Object]]
  },
  source:
    '<a href="http://twitter.com/download/iphone" rel="nofollow">Twitter for iPhone</a>',
  in_reply_to_status_id: null,
  in_reply_to_status_id_str: null,
  in_reply_to_user_id: null,
  in_reply_to_user_id_str: null,
  in_reply_to_screen_name: null,
  user: {
    id: 94733985,
    id_str: "94733985",
    name: "zonbi tamago",
    screen_name: "zonbitamago",
    location: "北海道はでっかいどう",
    description: "ぞんたま　zon bass trombone hokkaido university",
    url: "http: //t.co/8HrDj5V94l",
    entities: {
      url: [Object],
      description: [Object]
    },
    protected: false,
    followers_count: 104,
    friends_count: 116,
    listed_count: 5,
    created_at: "Sat Dec 05 05: 36: 45 +0000 2009",
    favourites_count: 212,
    utc_offset: null,
    time_zone: null,
    geo_enabled: false,
    verified: false,
    statuses_count: 2460,
    lang: "ja",
    contributors_enabled: false,
    is_translator: false,
    is_translation_enabled: false,
    profile_background_color: "EDECE9",
    profile_background_image_url:
      "http: //abs.twimg.com/images/themes/theme3/bg.gif",
    profile_background_image_url_https:
      "https: //abs.twimg.com/images/themes/theme3/bg.gif",
    profile_background_tile: false,
    profile_image_url:
      "http: //pbs.twimg.com/profile_images/928067133/STIL0017_normal.jpg",
    profile_image_url_https:
      "https: //pbs.twimg.com/profile_images/928067133/STIL0017_normal.jpg",
    profile_banner_url:
      "https: //pbs.twimg.com/profile_banners/94733985/1353817301",
    profile_link_color: "088253",
    profile_sidebar_border_color: "D3D2CF",
    profile_sidebar_fill_color: "E3E2DE",
    profile_text_color: "634047",
    profile_use_background_image: true,
    has_extended_profile: false,
    default_profile: false,
    default_profile_image: false,
    following: false,
    follow_request_sent: false,
    notifications: false,
    translator_type: "none"
  },
  geo: null,
  coordinates: null,
  place: null,
  contributors: null,
  retweeted_status: {
    created_at: "Fri Jun 22 11: 51: 50 +0000 2018",
    id: 1010128223022149600,
    id_str: "1010128223022149632",
    text: "息子よ...問題を読みなさい https: //t.co/qedW1jUMRK",
    truncated: false,
    entities: {
      hashtags: [],
      symbols: [],
      user_mentions: [],
      urls: [],
      media: [Array]
    },
    extended_entities: {
      media: [Array]
    },
    source: '<a href="http://www.echofon.com/" rel="nofollow">Echofon</a>',
    in_reply_to_status_id: null,
    in_reply_to_status_id_str: null,
    in_reply_to_user_id: null,
    in_reply_to_user_id_str: null,
    in_reply_to_screen_name: null,
    user: {
      id: 95228378,
      id_str: "95228378",
      name: "ERI",
      screen_name: "eri_sai",
      location: "Tokyo, JP",
      description: "Niigata / Portland, OR / Tokyo / 英語忘れそう/ 働く1児の母",
      url: null,
      entities: [Object],
      protected: false,
      followers_count: 1073,
      friends_count: 1983,
      listed_count: 11,
      created_at: "Mon Dec 07 16: 09: 42 +0000 2009",
      favourites_count: 652,
      utc_offset: null,
      time_zone: null,
      geo_enabled: true,
      verified: false,
      statuses_count: 8417,
      lang: "ja",
      contributors_enabled: false,
      is_translator: false,
      is_translation_enabled: false,
      profile_background_color: "FFF04D",
      profile_background_image_url:
        "http: //abs.twimg.com/images/themes/theme19/bg.gif",
      profile_background_image_url_https:
        "https: //abs.twimg.com/images/themes/theme19/bg.gif",
      profile_background_tile: false,
      profile_image_url:
        "http: //pbs.twimg.com/profile_images/444498056856104960/j-kvZZN9_normal.jpeg",
      profile_image_url_https:
        "https: //pbs.twimg.com/profile_images/444498056856104960/j-kvZZN9_normal.jpeg",
      profile_link_color: "0099CC",
      profile_sidebar_border_color: "FFF8AD",
      profile_sidebar_fill_color: "F6FFD1",
      profile_text_color: "333333",
      profile_use_background_image: true,
      has_extended_profile: false,
      default_profile: false,
      default_profile_image: false,
      following: false,
      follow_request_sent: false,
      notifications: false,
      translator_type: "none"
    },
    geo: null,
    coordinates: null,
    place: null,
    contributors: null,
    is_quote_status: false,
    retweet_count: 110236,
    favorite_count: 315764,
    favorited: true,
    retweeted: true,
    possibly_sensitive: false,
    lang: "ja"
  },
  is_quote_status: false,
  retweet_count: 110236,
  favorite_count: 0,
  favorited: true,
  retweeted: true,
  possibly_sensitive: false,
  lang: "ja"
};
