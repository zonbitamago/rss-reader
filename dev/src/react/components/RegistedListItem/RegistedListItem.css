.RegistedListItem {
}

.RegistedListItem-Text > h3 {
  color: #42a5f5;
  text-decoration: underline;
}
