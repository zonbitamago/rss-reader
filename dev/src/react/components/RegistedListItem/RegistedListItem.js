import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./RegistedListItem.css";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction";
import IconButton from "@material-ui/core/IconButton";
import Delete from "@material-ui/icons/Delete";

class RegistedListItem extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="RegistedListItem">
        <ListItem component="a" href={this.props.url}>
          <ListItemText
            className="RegistedListItem-Text"
            primary={this.props.text}
          />
          <ListItemSecondaryAction>
            <IconButton aria-label="Delete">
              <Delete />
            </IconButton>
          </ListItemSecondaryAction>
        </ListItem>
      </div>
    );
  }
}

RegistedListItem.propTypes = {
  text: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

RegistedListItem.defaultProps = {};

export default RegistedListItem;
