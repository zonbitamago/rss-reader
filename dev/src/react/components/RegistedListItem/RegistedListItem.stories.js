import React from "react";
import { storiesOf } from "@storybook/react";

import RegistedListItem from "./RegistedListItem";
import List from "@material-ui/core/List";

storiesOf("Atoms/RegistedListItem", module)
  .addDecorator(story => <List>{story()}</List>)
  .add("RegistedListItem", () => (
    <RegistedListItem name="sample" url="https://www.google.co.jp/" />
  ));
