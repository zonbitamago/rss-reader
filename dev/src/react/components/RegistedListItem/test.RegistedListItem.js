import React from "react";
import { shallow } from "enzyme";

import RegistedListItem from "./RegistedListItem";
import { electronUtil } from "../../../../__mocks__/electronUtilSettings";
import { store } from "../../../../__mocks__/storeSettings";

describe("RegistedListItem", () => {
  let component, props;

  beforeEach(() => {
    props = {
      name: "test",
      url: "https://google/co.jp",
      electronUtil: electronUtil,
      store: store
    };
    component = shallow(<RegistedListItem {...props} />);
  });

  it("should", () => {
    expect(component).toMatchSnapshot();
  });
});
