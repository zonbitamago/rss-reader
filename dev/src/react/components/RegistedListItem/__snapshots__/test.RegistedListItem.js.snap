// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistedListItem should 1`] = `
<div
  className="RegistedListItem"
>
  <WithStyles(ListItem)
    component="a"
    href="https://google/co.jp"
  >
    <WithStyles(ListItemText)
      className="RegistedListItem-Text"
      primary="test"
    />
    <WithStyles(ListItemSecondaryAction)>
      <WithStyles(IconButton)
        aria-label="Delete"
      >
        <pure(Delete) />
      </WithStyles(IconButton)>
    </WithStyles(ListItemSecondaryAction)>
  </WithStyles(ListItem)>
</div>
`;
